src/super_gradients/common/environment/env_helpers.py

```diff
@@ -59,24 +65,55 @@ def hydra_output_dir_resolver(ckpt_root_dir, experiment_name):
 
 def init_trainer():
     """
-    a function to initialize the super_gradients environment. This function should be the first thing to be called
-    by any code running super_gradients. It resolves conflicts between the different tools, packages and environments used
-    and prepares the super_gradients environment.
+    Initialize the super_gradients environment.
+
+    This function should be the first thing to be called by any code running super_gradients.
+    It resolves conflicts between the different tools, packages and environments used and prepares the super_gradients environment.
+    """
+
+    register_hydra_resolvers()
+
+    # We pop local_rank if it was specified in the args, because it would break
+    args_local_rank = pop_arg("local_rank", default_value=-1)
+
+    # Set local_rank with priority order (env variable > args.local_rank > args.default_value)
+    environment_config.DDP_LOCAL_RANK = int(os.getenv("LOCAL_RANK", default=args_local_rank))
+
+
+def register_hydra_resolvers():
+    """Register all the hydra resolvers required for the super-gradients recipes."""
+    OmegaConf.register_new_resolver("hydra_output_dir", hydra_output_dir_resolver, replace=True)
+
+
+def setup_gpu_mode(gpu_mode: MultiGPUMode = MultiGPUMode.OFF, num_gpus: int = None):
     """
-    OmegaConf.register_new_resolver(
-        "hydra_output_dir", hydra_output_dir_resolver, replace=True
-    )
+    If required, launch ddp subprocesses.
+    :param gpu_mode:    DDP, DP or Off
+    :param num_gpus:    Number of GPU's to use.
+    """
+    if require_gpu_setup(gpu_mode):
+        num_gpus = num_gpus or torch.cuda.device_count()
+        if num_gpus > torch.cuda.device_count():
+            raise ValueError(f"You specified num_gpus={num_gpus} but only {torch.cuda.device_count()} GPU's are available")
+        restart_script_with_ddp(num_gpus)
+
+
+def require_gpu_setup(gpu_mode: MultiGPUMode) -> bool:
+    """Check if the environment requires a setup in order to work with DDP."""
+    return (gpu_mode == MultiGPUMode.DISTRIBUTED_DATA_PARALLEL) and (not is_distributed())
+
+
+def pop_arg(arg_name: str, default_value: int = None) -> argparse.Namespace:
+    """Get the specified args and remove them from argv"""
+
     parser = argparse.ArgumentParser()
-    parser.add_argument("--local_rank", type=int, default=-1)  # used by DDP
+    parser.add_argument(f"--{arg_name}", default=default_value)
     args, _ = parser.parse_known_args()
 
-    # remove any flags starting with --local_rank from the argv list
-    to_remove = list(filter(lambda x: x.startswith('--local_rank'), sys.argv))
-    if len(to_remove) > 0:
-        for val in to_remove:
-            sys.argv.remove(val)
-
-    environment_config.DDP_LOCAL_RANK = args.local_rank
+    # Remove the ddp args to not have a conflict with the use of hydra
+    for val in filter(lambda x: x.startswith(f"--{arg_name}"), sys.argv):
```

Comment:
Yes I agree.

I thought about wrapping hydra.main with a function that would either:
- Ignore the args and run hydra.main, but with a warning to make the user see that his arg is being ignored by hydra
- Raise an error that would be more explicit than the error raise by hydra (which is very confusing when you are not used to it)

We could also instead add logs to make the user see that we droped "local_rank" from args when using the "pop_arg" function.

The last option we have is to just not support "python -m torch.distributed.launch --nproc_per_node=8 myscript.py" anymore (which uses local_rank as an arg).
In that case we would still support 2 ways to launch DDP:
1. Through pytorch with:
    - python -m torch.distributed.run --nproc_per_node=8 myscript.py (as opposed to torch.distributed.launch)
    - torchrun --nproc_per_node=8 myscript.py (this is just an alias for the previous line)
2. Through the use of the function "setup_gpu_mode"(that would set nproc_per_node either using a recipe or with a fixed value), and with a regular python call:
    - python myscript.py 
Note that we currently use "python -m torch.distributed.launch" and not "python -m torch.distributed.run" to use DDP.

Reply:
Did we verify that python -m torch.distributed.run --nproc_per_node=8 myscript.py works?
If so, then we shouldn't worry much about launch as it is already deprecated in torch. Also, the natural question that arrises is what's the difference between them regarding args?

Reply:
Yeah torch.distributed.run works.
I also tested without parsing local_rank and as expected torch.distributed.run still works in that case (local_rank is passed as env variable). Obviously in that case torch.distributed.launch does not work (local_rank passed as args).
I agree with you, but I feel like people might use torch.distributed.run and not understand the error(s), especially since  it worked until now